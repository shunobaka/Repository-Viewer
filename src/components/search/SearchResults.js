/**
 * @fileoverview Defines a SearchResults react component that displays a list of
 *    github users.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { Container, Row } from 'react-bootstrap';

import SearchResultItem from './SearchResultItem';

/**
 * SearchResults react component that displays a list of github users.
 * @param {object} props Contains an array of user information to be displayed
 */
const SearchResults = ({ users }) => {
  return (
    <Container>
      <Row>
        {users.map((user) => {
          return (
            <SearchResultItem key={user.id} user={user}></SearchResultItem>
          );
        })}
      </Row>
    </Container>
  );
};

SearchResults.propTypes = {
  users: PropTypes.array.isRequired,
};

export default SearchResults;
