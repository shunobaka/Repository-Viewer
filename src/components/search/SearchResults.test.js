import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import SearchResults from './SearchResults';

describe('searchResult snapshots', () => {
  const mockUsers = [
    {
      username: 'Test',
      id: 0,
      avatar: 'testImage',
    },
    {
      username: 'Test2',
      id: 1,
      avatar: 'testImage2',
    },
  ];

  it('multiple users renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <SearchResults users={mockUsers} />
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('single user renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <SearchResults users={[mockUsers[0]]} />
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('no users renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <SearchResults users={[]} />
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
