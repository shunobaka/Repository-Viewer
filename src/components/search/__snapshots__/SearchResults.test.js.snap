// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchResult snapshots multiple users renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-3"
    >
      <div
        className="jumbotron"
      >
        <a
          href="/user/Test"
          onClick={[Function]}
        >
          <h4
            className="text-center"
          >
            Test
          </h4>
          <img
            className=""
            src="testImage"
            width="100%"
          />
        </a>
      </div>
    </div>
    <div
      className="col-md-3"
    >
      <div
        className="jumbotron"
      >
        <a
          href="/user/Test2"
          onClick={[Function]}
        >
          <h4
            className="text-center"
          >
            Test2
          </h4>
          <img
            className=""
            src="testImage2"
            width="100%"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`searchResult snapshots no users renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  />
</div>
`;

exports[`searchResult snapshots single user renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-3"
    >
      <div
        className="jumbotron"
      >
        <a
          href="/user/Test"
          onClick={[Function]}
        >
          <h4
            className="text-center"
          >
            Test
          </h4>
          <img
            className=""
            src="testImage"
            width="100%"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
